src/__tests__/Button.test.js

```diff
@@ -267,39 +310,52 @@ describe("<Button />", () => {
   });
 
   it("disabled", () => {
-    const component = renderer.create(<Button disabled />).toJSON();
-    const instance = renderer.create(<Button disabled />).getInstance();
+    const B = renderer.create(<Button disabled />);
+
+    const component = B.toJSON();
+    const instance = B.root;
+
     const style = StyleSheet.flatten(component.props.style);
     expect(instance.props.disabled).toEqual(true);
     expect(style.backgroundColor).toEqual(rgba("#4630EB", 0.5));
   });
 
   it("opacity={0.5}", () => {
-    const instance = renderer.create(<Button opacity={0.5} />).getInstance();
+    const instance = renderer.create(<Button opacity={0.5} />).root;
     expect(instance.props.opacity).toEqual(0.5);
   });
 
-  it("ButtonType: highlight", () => {
-    const component = shallow(<Button highlight />);
-    expect(component.instance().props.highlight).toEqual(true);
-    expect(component.name()).toEqual("TouchableHighlight");
+  it("ButtonType: nativeFeedback", () => {
+    const component = shallow(<Button nativeFeedback />);
+
+    expect(component.props().nativeFeedback).toEqual(true);
+    expect(component.name()).toEqual("DummyTouchableNativeFeedback");
   });
 
+  /**
+   *  Apparently TouchableWithoutFeedback and TouchableHighlight needs a child to render
+   *  https://stackoverflow.com/questions/39862145/react-children-only-expected-to-receive-a-single-react-element-child-error-whe
+   */
+
   it("ButtonType: highlight", () => {
-    const component = shallow(<Button highlight />);
-    expect(component.instance().props.highlight).toEqual(true);
-    expect(component.name()).toEqual("TouchableHighlight");
-  });
+    const component = shallow(
+      <Button highlight>
+        <Button />
```

Comment:
We can add a Text here :) Button in Button is kinda weird. I web development is not allowed.